rpcs3/rpcs3qt/emu_settings.cpp

```diff
@@ -146,25 +152,59 @@ emu_settings::Render_Creator::Render_Creator()
 #endif
 
 #if defined(WIN32) || defined(HAVE_VULKAN)
-	// check for vulkan adapters
-	vk::context device_enum_context;
-	u32 instance_handle = device_enum_context.createInstance("RPCS3", true);
+	// Some drivers can get stuck when checking for vulkan-compatible gpus, f.ex. if they're waiting for one to get
+	// plugged in. This whole contraption is for showing an error message in case that happens, so that user has
+	// some idea about why the emulator window isn't showing up.
+	std::mutex mtx;
+	std::condition_variable cond;
+	std::unique_lock<std::mutex> lck(mtx, std::defer_lock);
+	std::atomic<bool> thread_running = true;
```

Comment:
this can be a simple non-atomic nor volatile bool instead if being used only inside the mutex region.